installer.sh

```diff
@@ -0,0 +1,2315 @@
+        
+        echo "XXX"
+        echo "40"
+        echo "Configuring Virtual Router..."
+        cmk update physicalnetwork state=Enabled id="$phy_id"
+        echo "XXX"
+        
+        local nsp_id=$(cmk list networkserviceproviders name=VirtualRouter physicalnetworkid="$phy_id" | jq -r '.networkserviceprovider[0].id')
+        local vre_id=$(cmk list virtualrouterelements nspid="$nsp_id" | jq -r '.virtualrouterelement[0].id')
+        
+        cmk configure virtualrouterelement enabled=true id="$vre_id"
+        cmk update networkserviceprovider state=Enabled id="$nsp_id"
+        
+        echo "XXX"
+        echo "50"
+        echo "Creating Pod..."
+        echo "XXX"
+        pod_id=$(cmk create pod name="$pod_name" \
+            zoneid="$zone_id" \
+            gateway="$GATEWAY" \
+            netmask="$NETMASK" \
+            startip="$pod_start" \
+            endip="$pod_end" | jq -r '.pod.id')
+        
+        [[ -z "$pod_id" ]] && error_exit "Failed to create pod"
+        
+        echo "XXX"
+        echo "60"
+        echo "Adding Cluster..."
+        echo "XXX"
+        cluster_id=$(cmk add cluster \
+            zoneid="$zone_id" \
+            podid="$pod_id" \
+            clustername="$cluster_name" \
+            clustertype=CloudManaged \
+            hypervisor=KVM | jq -r '.cluster[0].id')
+
+        [[ -z "$cluster_id" ]] && error_exit "Failed to add cluster"
+        
+        echo "XXX"
+        echo "70"
+        echo "Adding Host..."
+        echo "XXX"
+        # Add Host
+        cmk add host zoneid="$zone_id" \
+            podid="$pod_id" \
+            clusterid="$cluster_id" \
+            hypervisor=KVM \
+            username=root \
+            password="$root_pass" \
+            url="http://$KVM_HOST_IP"
+        
+        echo "XXX"
+        echo "80"
+        echo "Adding Primary Storage..."
+        echo "XXX"
+        # Add Primary Storage
+        cmk create storagepool name="$primary_name" \
+            zoneid="$zone_id" \
+            podid="$pod_id" \
+            clusterid="$cluster_id" \
+            url="nfs://$nfs_server$primary_path" \
+            hypervisor=KVM \
+            scope=zone
+        
+        echo "XXX"
+        echo "90"
+        echo "Adding Secondary Storage..."
+        echo "XXX"
+        # Add Secondary Storage
+        cmk add imagestore name="$secondary_name" \
+            zoneid="$zone_id" \
+            url="nfs://$nfs_server$secondary_path" \
+            provider=NFS
+
+        echo "XXX"
+        echo "95"
+        echo "Enabling Zone..."
+        echo "XXX"
+        cmk update zone allocationstate=Enabled id="$zone_id"
+        
+        echo "XXX"
+        echo "100"
+        echo "Zone deployment completed successfully!"
+        echo "XXX"
+    } | dialog --backtitle "$SCRIPT_NAME" \
+               --title "Zone Deployment" \
+               --gauge "Deploying CloudStack Zone..." 10 70 0
+
```

Comment:
Update each step in deploy zone into smaller methods?
- Add Traffic Types
- Add Public IP Range
- Add Pod
- Add Cluster
- Add Host
- Add Primary Storage
- Add Secondary Storage